diff --git a/notification.c b/notification.c
index 70a01f6..ca01953 100644
--- a/notification.c
+++ b/notification.c
@@ -356,7 +356,7 @@ int notification_init(notification * n, int id)
 
         if (n->progress) {
                 char pg[10];
-                sprintf(pg, "[%3d%%]", n->progress - 1);
+                sprintf(pg, "%3d%%", n->progress - 1);
                 n->msg = string_replace_all("%p", pg, n->msg);
         } else {
                 n->msg = string_replace_all("%p", "", n->msg);
@@ -462,6 +462,7 @@ int notification_init(notification * n, int id)
 
         if (strlen(n->msg) == 0) {
                 notification_close(n, 2);
+                notification_run_script(n);
                 printf("skipping notification: %s %s\n", n->body, n->summary);
         } else {
                 g_queue_insert_sorted(queue, n, notification_cmp_data, NULL);
